lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1549,9 +1549,19 @@ static bool performCompileStepsPostSILGen(CompilerInstance &Instance,
     SerializationOptions serializationOpts =
         Invocation.computeSerializationOptions(outs, Instance.getMainModule());
 
+    // Infer if this is an emit-module job part of an incremental build,
+    // vs a partial emit-module job (with primary files) or other kinds.
+    // We may want to rely on a flag instead to differentiate them.
+    const bool isEmitModuleSeparately =
+        Action == FrontendOptions::ActionType::EmitModuleOnly &&
+        MSF.is<ModuleDecl *>() &&
+        Instance.getInvocation()
```

Comment:
Curious why this depends on the state of function body skipping.

Reply:
The flag controlling `NonInlinableWithoutTypes` is used here as an heuristic to distinguish incremental `EmitModuleOnly` jobs from the classic non-incremental ones. I'd rather have a flag specifically about incremental builds but IIUC that flag was intentionally removed.

Reply:
@CodaFi would you prefer if the compiler emitted incremental information for all emit-module jobs? Outside of emit-module-separately, these are mostly used in testing and to build core library as far as I know.

Reply:
I think it certainly wouldn't hurt. The cost of embedding Swiftdeps is negligible.